hpx/lcos/async.hpp

```diff
@@ -158,43 +193,23 @@ namespace hpx { namespace detail
         }
     };
 
-    ///////////////////////////////////////////////////////////////////////////
+    // BOOST_SCOPED_ENUM(launch)
     template <typename Policy>
     struct async_dispatch<Policy,
```

Comment:
The file `hpx/lcos/async.hpp` is included by `hpx/async.hpp`. If we move this definition to `hpx/async.hpp` it will not be available for the stuff needed in `hpx/lcos/async.hpp`.